osbs/repo_utils.py

```diff
@@ -60,10 +60,73 @@ def __init__(self, dir_path='', file_name=REPO_CONFIG_FILE):
             with open(file_path) as f:
                 self.container = (yaml.load(f) or {})
 
+        modules = self.container.get('compose', {}).get('modules', [])
+
+        self.container_module_specs = []
+        value_errors = []
+        for module in modules:
+            try:
+                self.container_module_specs.append(ModuleSpec.from_text(module))
+            except ValueError as e:
+                value_errors.append(e)
+        if value_errors:
+            raise ValueError(value_errors)
+
     def is_autorebuild_enabled(self):
         return self._config_parser.getboolean('autorebuild', 'enabled')
 
 
+class ModuleSpec(object):
+    """
+    Specification for a to be requested module.
+
+    This module representation is simplified from the possible
+    NAME:STREAM:VERSION:CONTEXT:ARCH/PROFILE by not supporting ARCH, which
+    should be determined by the architecture of the build, and by not
+    supporting partal specifications such as NAME:::CONTEXT.
+    """
+
+    def __init__(self, name, stream, version=None, context=None, profile=None):
+        self.name = name
+        self.stream = stream
+        self.version = version
+        self.context = context
+        self.profile = profile
+
+    def to_str(self, include_profile=True):
+        result = self.name + ':' + self.stream
+        if self.version:
+            result += ':' + self.version
+        if self.context:
+            result += ':' + self.context
+        if include_profile and self.profile:
+            result += '/' + self.profile
+
+        return result
+
+    def __repr__(self):
+        return "ModuleSpec({})".format(self.to_str())
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+    @classmethod
+    def from_text(cls, text):
```

Comment:
Nit: if this is the inverse of `to_str` should it be called `from_str`?

Reply:
Done!